src/main/java/org/apache/commons/io/IOUtils.java

```diff
@@ -414,6 +433,48 @@ public static void close(final Closeable closeable, final IOConsumer<IOException
         }
     }
 
+    /**
+     * Closes the entries in the given {@link Stream} as null-safe operations,
+     * and closes the underlying {@code Stream}.
+     *
+     * @param <T> The element type.
+     * @param consumer Consume the IOException thrown by {@link Closeable#close()}.
+     * @param closeables The resource(s) to close, may be null.
+     * @throws IOException if an I/O error occurs.
+     */
+    public static <T extends Closeable> void close(final IOConsumer<IOException> consumer, final Stream<T> closeables) throws IOException {
+        if (closeables != null) {
+            try {
+                close(closeables);
+            } catch (final IOException e) {
+                if (consumer != null) {
+                    consumer.accept(e);
+                }
+            } finally {
+                closeables.close();
```

Comment:
This seems wrong to me since this method did not create the stream. The caller should manage the stream IMO, it does not matter that a stream also happens to be closeable.

Reply:
Agreed, closing the stream here is making an assumption that this should be a terminal operation on the stream, which is a limitation on its usefulness. The caller can close the stream on their own when they are finished with it. I will update the code.

Reply:
Please, do. And also, add a single sentence to the Javadocs, that mentions the fact. (Always good, if the caller is made aware of his/her ressponsibilities.)